source/server/active_stream_socket.h

```diff
@@ -0,0 +1,222 @@
+                       Network::ListenerFilterPtr&& filter) override {
+    accept_filters_.emplace_back(
+        std::make_unique<GenericListenerFilter>(listener_filter_matcher, std::move(filter)));
+  }
+
+  // Network::ListenerFilterCallbacks
+  Network::ConnectionSocket& socket() override { return *socket_.get(); }
+  Event::Dispatcher& dispatcher() override;
+  void continueFilterChain(bool success) override;
+  void setDynamicMetadata(const std::string& name, const ProtobufWkt::Struct& value) override;
+  envoy::config::core::v3::Metadata& dynamicMetadata() override {
+    return stream_info_->dynamicMetadata();
+  };
+  const envoy::config::core::v3::Metadata& dynamicMetadata() const override {
+    return stream_info_->dynamicMetadata();
+  };
+
+  StreamInfo::FilterState& filterState() override { return *stream_info_->filterState().get(); }
+
+  ActiveStreamListenerBase& listener_;
+  Network::ConnectionSocketPtr socket_;
+  const bool hand_off_restored_destination_connections_;
+  std::list<ListenerFilterWrapperPtr> accept_filters_;
+  std::list<ListenerFilterWrapperPtr>::iterator iter_;
+  Event::TimerPtr timer_;
+  std::unique_ptr<StreamInfo::StreamInfo> stream_info_;
+  bool connected_{false};
+};
+
+using ActiveInternalSocket = ActiveStreamSocket;
+
+class ActiveStreamListenerBase : public ActiveListenerImplBase {
+public:
+  ActiveStreamListenerBase(Network::ConnectionHandler& parent, Event::Dispatcher& dispatcher,
+                           Network::ListenerPtr&& listener, Network::ListenerConfig& config);
+  static void emitLogs(Network::ListenerConfig& config, StreamInfo::StreamInfo& stream_info);
+
+  virtual void incNumConnections() PURE;
+  virtual void decNumConnections() PURE;
+  virtual Event::Dispatcher& dispatcher() PURE;
+
+  /**
+   * Create a new connection from a socket accepted by the listener.
+   */
+  virtual void newConnection(Network::ConnectionSocketPtr&& socket,
+                             std::unique_ptr<StreamInfo::StreamInfo> stream_info) PURE;
+
+  virtual Network::BalancedConnectionHandlerOptRef
+  getBalancedHandlerByAddress(const Network::Address::Instance& address) PURE;
+
+  void onSocketAccepted(std::unique_ptr<ActiveStreamSocket> active_socket) {
+    // Create and run the filters
+    config_->filterChainFactory().createListenerFilterChain(*active_socket);
+    active_socket->continueFilterChain(true);
+
+    // Move active_socket to the sockets_ list if filter iteration needs to continue later.
+    // Otherwise we let active_socket be destructed when it goes out of scope.
+    if (active_socket->iter_ != active_socket->accept_filters_.end()) {
+      active_socket->startTimer();
+      LinkedList::moveIntoListBack(std::move(active_socket), sockets_);
+    } else {
+      // If active_socket is about to be destructed, emit logs if a connection is not created.
+      if (!active_socket->connected_) {
+        emitLogs(*config_, *active_socket->stream_info_);
```

Comment:
It seems like some error handling was lost as part of this refactor.  What happened?

https://github.com/envoyproxy/envoy/blob/fff2cf819917ae54d7170cacad179e75382a1e25/source/server/active_tcp_listener.cc#L267